app/validate_txs.go

```diff
@@ -122,3 +144,15 @@ func encodeBlobTxs(blobTxs []*tx.BlobTx) [][]byte {
 	}
 	return txs
 }
+
+// countOccurrence takes a strings slice and counts the number
+// of time the provided item exists in that slice.
+func countOccurrence(slice []string, item string) int {
+	count := 0
+	for _, v := range slice {
+		if v == item {
+			count++
+		}
+	}
+	return count
+}
```

Comment:
very very optional

we might be able to refactor this and `msgTypes` into a single function that returns all of the different msg types in a map or smth

Reply:
we will need to sequentially go over the message types  in all cases, same complexity. Or you mean we make them part of a single function?

Reply:
> Or you mean we make them part of a single function?

yeah, just this. very optional

Reply:
[54583ae](https://github.com/celestiaorg/celestia-app/pull/3942/commits/54583ae722bffa9a4c6756c17b3b00278ded1f3b)